server.js

```diff
@@ -108,11 +109,31 @@ const cleanRedirect = (url) => (url.startsWith("/") ? url : "");
 api.get(/^\/simulator(.*)$/, ({ url }, res) => res.redirect(`/api/textile${cleanRedirect(url)}`));
 
 // Note: Text/JSON request body parser (JSON is decoded in Elm)
-api.all(/(.*)/, bodyParser.json(), (req, res) => {
+api.all(/(.*)/, bodyParser.json(), async (req, res) => {
+  let headers = {};
+  if (req.headers.token) {
+    headers["token"] = req.headers.token;
+  }
+  let processes;
+  if (process.env.NODE_ENV == "test") {
+    headers["fakeDetails"] = "true";
+  }
+  try {
+    const processesUrl = `http://127.0.0.1:${django_port}/processes/processes.json`;
+    const processesRes = await fetch(processesUrl, { headers: headers });
```

Comment:
So that means one HTTP request is issued to the Django instance (which believes performs at least one SQL request on every render to check for user authentication status) per incoming HTTP request (we had 500k API hits the other day)… will this scale well? Isn't there any other way to preload the two versions of the processes (with and without details)? And/or use memoization?

Reply:
Not sure if we should access django directly through 8002 or let nginx or the parcel proxy forward the requests.

Reply:
I don't know the current status, but the detailed impact could be retrieved once upon authentication, then stored in memory on the elm side. Is this the case currently?

Reply:
I agree this would be a better strategy, yes.